### PR TITLE
[Table Model] Create table device

### PR DESCRIPTION
**Pull request by MarcosZyk**
## Description

Support Create Table Device, including:
1. Statement CreateDevice
2. PlanNode CreateTableDeviceNode, including ser/deser and split logic
3. LogicalPlan stage logic 
4. execution logic invoking SchemaRegion

### Local execution result
image

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  }
+
+  @Override
+  public List<WritePlanNode> splitByPartition(IAnalysis analysis) {
+    Map<TRegionReplicaSet, List<Integer>> splitMap = new HashMap<>();
+    for (int i = 0; i < devicePathList.size(); i++) {
+      TRegionReplicaSet regionReplicaSet =
+          analysis
+              .getSchemaPartitionInfo()
+              .getSchemaRegionReplicaSet(new StringArrayDeviceID(devicePathList.get(i).getNodes()));
```
Comment (JackieTien97) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java:192]: pass String database and use `IDeviceID.Factory.create(String[])` to create IDeviceId.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/LogicalPlanner.java [hunk 2]
```diff
@@ -181,4 +192,37 @@ private enum Stage {
...
+    context.setQueryType(QueryType.WRITE);
+    analysis.setStatement(statement);
+
+    PathPatternTree patternTree = new PathPatternTree();
+    for (PartialPath devicePath : devicePathList) {
+      patternTree.appendFullPath(devicePath.concatNode(ONE_LEVEL_PATH_WILDCARD));
+    }
+    SchemaPartition partition =
+        partitionFetcher.getOrCreateSchemaPartition(
+            patternTree, context.getSession().getUserName());
```
[the remark this answers is not in the document]
Comment (JackieTien97) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/LogicalPlanner.java:222]: String databaseName, List<IDeviceId>

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java (hunk 1 above)
Comment (JackieTien97) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java:192]: add new method in getSchemaRegionReplicaSet(String database, IDeviceId)

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/LogicalPlanner.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (MarcosZyk) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/LogicalPlanner.java]: Already set interface in Metadata, but a todo is left waiting for implementation.

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class CreateTableDeviceNode extends WritePlanNode {
+  private final List<PartialPath> devicePathList;
```
[the remark this answers is not in the document]
Comment (MarcosZyk) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java]: Fixed.

[the remark this answers is not in the document]
Comment (MarcosZyk) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java]: The db name and table name have been separated from the device id, Object[].

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java (hunk 1 above)
Comment (MarcosZyk) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/relational/planner/node/CreateTableDeviceNode.java]: Fixed implementation.

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**